internal/rbd/nodeserver.go

```diff
@@ -287,14 +345,22 @@ func (ns *NodeServer) stageTransaction(ctx context.Context, req *csi.NodeStageVo
 	}
 	// Mapping RBD image
 	var devicePath string
-	devicePath, err = attachRBDImage(ctx, volOptions, cr)
+	devicePath, err = attachRBDImage(ctx, volOptions, devicePath, cr)
 	if err != nil {
 		return transaction, err
 	}
 	transaction.devicePath = devicePath
+
 	util.DebugLog(ctx, "rbd image: %s/%s was successfully mapped at %s\n",
 		req.GetVolumeId(), volOptions.Pool, devicePath)
 
+	if volOptions.Mounter == rbdNbdMounter {
+		err = updateRBDImageMetadataStash(req.GetStagingTargetPath(), devicePath)
```

Comment:
as in this process we are just reattaching do we still to continue with other operations in `stageTransaction` functions like

* Kernel check and flattening
* Proress encryption
* StagingPath creation
* Mount volume to stagingpath
* chmod 

Because failure of some of these operations can undo things also

Reply:
As per our discussion, this looks good!